design_utils/utils.py

```diff
@@ -16,7 +19,239 @@
 from numpy import genfromtxt
 
 from aposteriori.config import MAKE_FRAME_DATASET_VER, UNCOMMON_RESIDUE_DICT
-from aposteriori.data_prep.create_frame_data_set import DatasetMetadata
+from aposteriori.data_prep.create_frame_data_set import (
+    DatasetMetadata,
+)
+
+
+def constrain_aa_resnum(res_to_fix):
+    """
+    At the moment the user inputs for residue fixing and residue prediction are taken as a string.
+    This function takes a user input and separates the integer bits from string bits. Particularly useful if the user wants to specify the aminoacid constraints when fixing residues.
+    For example, if the user puts --res_to_fix "10Y", this function will come in handy to constrain the 10th residue to a tyrosine.
+    If the user enters --res_to_fix "5", then the aminoacid will be considered "X" at the beginning, but this will not be inserted into the predictions function to create dictionaries for individual chains of proteins with their residue numbers.
```

Comment:
This docstring does not explain what the function does. 

"At the moment" may change in the future and in fact, if anything changes it will confuse the user. Have a look at the other docstrings. Your docstring should describe what your function does with the input and what it returns.